docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+**A valid block MUST contain an eligible agenda.**. In other words, if there is no eligible agenda, there is no block progress in the blockchain.
+
+In addition to the agenda, the leader can also include some other transactions.
+
+1. RecordChatLog (chatting logs for the very height): this doesn't change the state, but is used as the governance minutes for the agenda.
+2. ReportMisbehavior (double votes in the consensus and chat finalization misbehaviors by a past consensus leader): this will automatically lead to the slashing of the voting power of the misbehaving validator.
+
+These two transactions are the only exceptions that are not part of the agenda.
+
+## Consensus Leader
+
+You might notice that the role of the consensus leader is very important in the simperby governance.
+
+### Responsibilities
+
+The leader should be **responsible**.
+
+- The leader should be online for the whole round, providing enough frequency of chat chain semifinalizations.
+- The leader should propose a block as soon as possible if there is any eligible agenda.
+- The leader should report misbehaviors as well.
+
+Again, one of the key requirements of Simperby is making the node operation as lightweight as possible, especially in terms of how many times and how long the node should be online. Thus laying the burden of being a leader on a round robin, which eventually involves all the validators evenly, will just exhaust them.
+
+To mitigate this problem, Simperby supports a **stable leader** feature.
+
+- The leader order is determined in the state.
+- The leader order doesn't change throughout the heights if the order specified in the state stays the same.
+(i.e., the leader for the first round stays the same)
+- For a single height, it is also possible to repeat the first few leaders for several rounds. (e.g., `1->1->1->2->2->3->4->...` instead of `1->2->3->4->...`)
+
+Another advantage of a stable leader is that it is **predictable**. Every validator can be quite sure how often they should turn on their node based on the order in the stable leader list.
+
+As long as the selected first few leaders are honest(not byzantine), and *faithful* (fulfilling their responsibilities), the network will be **live**.
+
+### Authority
+
+The leader has **authority**.
+
+- The leader can choose which agenda to include in the block if there are multiple eligible agendas.
+- The leader can choose which chat chain to semifinalize.
+
+A *good* node should be objective to agendas; they should choose the first eligible agenda, instead of waiting until it observes another one in their personal favor. Similarly, a 'good' node should be objective to the chat chain semifinalization as well.
+
+As long as the selected first few leaders are honest, faithful, and even good,
+the network will be not only **live** but also **fair**. (note that it's always **safe** regardless of the leader's behavior; safety is guaranteed by the BFT consensus)
+
+### So What?
```

Comment:
Let's use more meaningful words for the title instead of context-depending words.
(e.g. dealing with a lazy or bad leader)

Reply:
updated